lib/mrtrix3/run.py

```diff
@@ -335,3 +338,52 @@ def versionMatch(item):
 
   app.error('Unable to find executable for MRtrix3 command ' + item)
 
+
+
+# If the target executable is not a binary, but is actually a script, use the
+#   shebang at the start of the file to alter the subprocess call
+def _shebang(item):
+  import os
+  from mrtrix3 import app
+  from distutils.spawn import find_executable
+  # If a complete path has been provided rather than just a file name, don't perform any additional file search
+  if os.sep in item:
+    path = item
+  else:
+    path = versionMatch(item)
+    if path == item:
+      path = find_executable(exeName(item))
+  if not path:
+    app.debug('File \"' + item + '\": Could not find file to query')
+    return []
+  # Read the first 1024 bytes of the file
+  with open(path, 'rb') as f:
+    data = f.read(1024)
+  # List of permissible text characters
+  textchars = bytearray({7,8,9,10,12,13,27} | set(range(0x20, 0x100)) - {0x7f})
+  # Are there any non-text characters? If so, it's a binary file, so don't go looking for a shebang
+  if data.translate(None, textchars):
+    app.debug('File \"' + item + '\": Not a text file')
+    return []
+  # Try to find the shebang line
+  for line in data.splitlines():
+    line = line.strip()
+    if len(line) > 2 and line[0:2] == '#!':
+      shebang = line[2:].split(' ')
+      if app.isWindows():
+        # On Windows, /usr/bin/env can't be easily found, and any direct interpreter path will have a similar issue.
+        #   Instead, manually find the right interpreter to call using distutils
+        if shebang[0] == '/usr/bin/env':
```

Comment:
Yes, I suppose if somebody adds a shebang pointing explicitly to some other executable called '`env`' they probably deserve what's coming to them.